cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -2182,5 +2185,24 @@ TEST(TestRankForFixedSizeBinary, RankFixedSizeBinary) {
       ArrayFromJSON(binary_type, R"(["aaa", "   ", "eee", null, "eee", null, "   "])"));
 }
 
+
+TEST(TestRankForReal, RankRealChunked) {
```

Comment:
Here I have created the Chunk test as a separate item, which duplicates all of the looping logic. If we didn't want to do that, could also just implement the ChunkedArray constructors in the existing tests. 

I think most tests go with this pattern, so happy to continue and copy over all of the rest of the type tests if that's what we want